### PR TITLE
Milestone, BannerInstructionMilestone, RouteMilestone, StepMilestone, VoiceInstructionMilestone convert to Kotlin

### PR DESCRIPTION
**Pull request by Lebedinsky**
### Description

Migrate `TriggerMilestone`, `BannerInstructionMilestone`, `RouteMilestone`, `StepMilestone`, `VoiceInstructionMilestoneProperty` to Kotlin [issue](https://github.com/mapbox/navigation-sdks/issues/169)

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

## Testing

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/Milestone.kt [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress
+
+/**
+ * Base Milestone statement. Subclassed to provide concrete statements.
+ *
+ * @since 0.4.0
+ */
+abstract class Milestone
+// Public exposed for creation of milestone classes outside SDK
+constructor(private val builder: Builder) {
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/Milestone.kt]: Could this be a refactored into primary constructor?

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/BannerInstructionMilestone.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import java.util.ArrayList
+
+/**
+ * A default milestone that is added to [com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation]
+ * when default milestones are enabled.
+ *
+ * Please note, this milestone has a custom trigger based on location progress along a route.  If you
+ * set custom triggers, they will be ignored in favor of this logic.
+ */
+class BannerInstructionMilestone private constructor(
```
Comment (RingerJK) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/BannerInstructionMilestone.kt]:
```suggestion
class BannerInstructionMilestone 
private constructor(
```

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/Milestone.kt [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+     * Build a new [Milestone]
+     *
+     * @since 0.4.0
+     */
+    abstract class Builder {
+
+        private var identifier: Int = 0
+        private var instruction: Instruction? = null
+
+        // internal abstract val trigger: Trigger.Statement?
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/Milestone.kt]: Could we remove this commented out code?

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/StepMilestone.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import com.mapbox.services.android.navigation.v5.utils.extensions.ifNonNull
+
+/**
+ * Using a Step Milestone will result in
+ * [MilestoneEventListener.onMilestoneEvent]
+ * being invoked every step if the condition validation returns true.
+ *
+ * @since 0.4.0
+ */
+class StepMilestone private constructor(
```
Comment (RingerJK) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/StepMilestone.kt]:
```suggestion
class StepMilestone 
private constructor(
```

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/Milestone.kt (hunk 1 above)
Comment (Lebedinsky) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/Milestone.kt]: It's already primary)

**Pull request merged**